app_dart/lib/src/service/build_status_provider.dart

```diff
@@ -22,36 +22,50 @@ class BuildStatusProvider {
 
   final DatastoreServiceProvider datastoreProvider;
 
+  @visibleForTesting
+
+  /// This is how far back this will reference to calculate the build status.
+  static const int numberOfCommitsToReference = 20;
+
   /// Calculates and returns the "overall" status of the Flutter build.
   ///
   /// This calculation operates by looking for the most recent success or
   /// failure for every (non-flaky) task in the manifest.
   Future<BuildStatus> calculateCumulativeStatus() async {
     final Map<String, bool> checkedTasks = <String, bool>{};
-    bool isLatestBuild = true;
+    bool isLatestCommitStatus = true;
 
     await for (CommitStatus status in retrieveCommitStatus()) {
```

Comment:
I have to wrap around my head on this section. Is there a way to simply it?

Reply:
I think inherently this is hard algorithm to follow. The hardest and most time consuming part of this PR was just understanding the PR. I have two ideas for making it a bit more self explanatory.

1. Renaming:
  a. `!checked` => `taskIsInProgress`
  b. `checkedTask` => `tasksInProgress`
2. ASCII diagrams of a build dashboard. I think this is the most beneficial as it's what got me to understand the algorithm.

What do you think?

Reply:
Those are helpful! In addition, we could reduce the nesting ifs with "early returns", see https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Reply:
Thanks for the resource! I was able to extract the latest commit status logic to its own block which seems to have cleaned up most of the complexity.